diffkemp/simpll/passes/SeparateCallsToBitcastPass.cpp

```diff
@@ -0,0 +1,138 @@
+                        } else {
+                            // Bitcast the argument so that the types match.
+                            auto newArg = CastInst::Create(
+                                    Instruction::BitCast,
+                                    *arg,
+                                    *paramType,
+                                    "",
+                                    newArgs.empty() ? cast<Instruction>(Call)
+                                                    : cast<Instruction>(
+                                                            newArgs.back()));
+
+                            newArg->setDebugLoc(Call->getDebugLoc());
+                            newArgs.push_back(newArg);
+                        }
+
+                        ++paramType;
+                        ++arg;
+                    }
+
+                    // Add the remaining arguments if there are any.
+                    while (arg != endArg) {
+                        newArgs.push_back(*arg);
+                        ++arg;
+                    }
+
+                    // Create a new call instruction using the
+                    // source function and bitcasted arguments.
+                    auto newCall = CallInst::Create(srcFun, newArgs, "", Call);
+                    newCall->setDebugLoc(Call->getDebugLoc());
+                    Instruction *replacementValue = newCall;
+
+                    if (Call->getType() != newCall->getType()
+                        && !newCall->getType()->isVoidTy()) {
+                        // If return types do not match, bitcast the new
+                        // call result to the original result type.
+                        // Calls with a void return type are not bitcasted.
+                        auto returnBitCast =
+                                CastInst::Create(Instruction::BitCast,
+                                                 newCall,
+                                                 Call->getType(),
+                                                 "",
+                                                 Call);
+
+                        returnBitCast->setDebugLoc(Call->getDebugLoc());
+                        replacementValue = returnBitCast;
+                    }
+
+                    // Replace the old call instruction with the last generated
+                    // instruction.
+                    DEBUG_WITH_TYPE(DEBUG_SIMPLL,
+                                    dbgs() << "Replacing :" << *Call
+                                           << "\n   with :" << *replacementValue
+                                           << "\n");
+                    Call->replaceAllUsesWith(replacementValue);
+                    toRemove.push_back(Call);
+                }
+            }
+        }
+    }
+
+    // Remove replaced call instructions.
+    for (auto Call : toRemove) {
+        Call->eraseFromParent();
+    }
+    toRemove.clear();
```

Comment:
I think the clearing of the vector can be omitted, since it is done automatically when the variables on the stack are destroyed on leaving the function.

Reply:
Alright, will do.